client/state/sites/reducer.js

```diff
@@ -98,18 +99,38 @@ export function items( state = {}, action ) {
 			return state;
 		}
 
-		case SITE_RECEIVE: {
-			const site = pick( action.site, VALID_SITE_KEYS );
-			return Object.assign( {}, state, {
-				[ site.ID ]: site
-			} );
-		}
-
+		case SITE_RECEIVE:
 		case SITES_RECEIVE:
-			return action.sites.reduce( ( memo, site ) => {
-				memo[ site.ID ] = pick( site, VALID_SITE_KEYS );
+		case SITES_UPDATE:
+			// Normalize incoming site(s) to array
+			const sites = action.site ? [ action.site ] : action.sites;
```

Comment:
let's be careful here. the case statement doesn't create a new scoping block and so `const sites` can only exist in this branch.